src/components/Calendar/doubleCalendar/index.js

```diff
@@ -0,0 +1,350 @@
+    normalizeDate,
+    getFirstDayMonth,
+    addMonths,
+    getNextFocusedDate,
+    isSameMonth,
+    isEmptyRange,
+    isDateBelowLimit,
+} from '../helpers';
+import {
+    useYearsRange,
+    useLastDayMonth,
+    useMoveFocusedDayFunction,
+    useMoveFocusedMonthFunction,
+} from './hooks';
+import StyledControlsContainer from '../styled/controlsContainer';
+import StyledArrowButton from '../styled/arrowButton';
+import StyledMonthsContainer from './styled/monthsContainer';
+import StyledTable from '../styled/table';
+import StyledContainer from './styled/container';
+import StyledCalendar from './styled/calendar';
+import StyledDivider from './styled/divider';
+import MonthHeader from './monthHeader';
+
+export default function DoubleCalendar(props) {
+    const {
+        id,
+        className,
+        style,
+        locale,
+        value,
+        minDate,
+        maxDate,
+        onChange,
+        selectedRange,
+        selectionType,
+    } = props;
+    const [focusedDate, setFocusedDate] = useState(normalizeDate(value));
+    const [currentMonth, setCurrentMonth] = useState(getFirstDayMonth(normalizeDate(value)));
+    const [currentRange, setCurrentRange] = useState(selectedRange);
+    const [enableNavKeys, setEnableNavKeys] = useState(false);
+
+    const rightCalendarMonth = addMonths(currentMonth, 1);
+    const firstMonthLabelId = useUniqueIdentifier('first-month');
+    const secondMonthLabelId = useUniqueIdentifier('second-month');
+    const currentYear = currentMonth.getFullYear();
+    const rightCalendarYear = rightCalendarMonth.getFullYear();
+    const yearsRange = useYearsRange(minDate, maxDate, currentMonth);
+    const lastYearItem = yearsRange[yearsRange.length - 1];
+    const minSelectableDate = minDate || new Date(yearsRange[0].value, 0, 1);
+    const maxSelectableDate = maxDate || new Date(lastYearItem.value, 11, 31);
+    const lastDayMonth = useLastDayMonth(currentMonth);
+    const disableNextMonth = addMonths(rightCalendarMonth, 1) > maxSelectableDate;
+    const disablePreviousMonth = lastDayMonth < minSelectableDate;
+
+    const moveFocusedDay = useMoveFocusedDayFunction(focusedDate, currentMonth, minDate, maxDate);
+    const moveFocusedMonth = useMoveFocusedMonthFunction(focusedDate, minDate, maxDate);
+
+    const handleOnDayFocus = () => setEnableNavKeys(true);
+    const handleOnDayBlur = () => setEnableNavKeys(false);
+    const handleOnDayHover = useCallback(
+        date => {
+            if (selectionType === 'single' || isEmptyRange(selectedRange)) return;
+
+            const [rangeStart, rangeEnd] = selectedRange;
+            if (rangeEnd) return;
+
+            const [currentRangeStart] = currentRange;
+
+            if (isDateBelowLimit(date, rangeStart)) {
+                setCurrentRange([currentRangeStart]);
+            } else {
+                setCurrentRange([rangeStart, date]);
+            }
+        },
+        [currentRange, selectedRange, selectionType],
+    );
```

Comment:
It is good practice in large components like this, put all hook calls in custom hook with descriptive names so that the code is much more readable.